Add EditableText scroll into view breaking change note (#8916)
flutter/website: src/release/breaking-changes/editable-text-scroll-into-view.md

---
title: Updated EditableText scroll into view behavior
description: >
  Improve EditableText selection scroll into view behavior to always
  use the current selection extent.
---

## Summary

> `Editable.onCaretChanged` callback is removed and with this change,
> `EditableText` behavior with regard to scrolling the selection into view
> changes.

Reviewer: ```suggestion
The `Editable.onCaretChanged` callback is removed. With this change,
`EditableText` behavior for scrolling the selection into view
changes.
```

## Context

> To date, `EditableText` has used multiple mechanisms ensuring the selection
> extent or caret is scrolled into view on updates. 

Reviewer: Not sure what you mean by "on updates".

```suggestion
Previously, `EditableText` used multiple mechanisms to determine the
extent of the selection, or to ensure that the caret is scrolled into view.
```

Author: Thanks for the edits. Yes, this is a bit convoluted: what I meant were the mechanisms used on scrolling into view (which happens on user updates) either of:
- selection extent (non-collapsed selection)
- caret position (collapsed selection).

Let me reword as:

> Previously, upon scrolling into view to show user updates, `EditableText` used multiple mechanisms to determine the
> extent of the selection or the caret location.

## Description of change

> By removing the `Editable.onCaretChanged` callback, `EditableText` will always
> use the most up to selection extent location when scrolling to show it.
> Specifically this results in improved scroll into view behavior after
> changing selection from collapsed to non-collapsed via
> `userUpdateTextEditingValue()`.

Reviewer: What does "use the most up to selection extent location" mean?
I've made some tweaks, but would do more once I understand this phrase.

```suggestion
By removing the `Editable.onCaretChanged` callback, `EditableText` will always
use the most up to selection extent location when scrolling to show it.
Specifically, this improves scroll into view behavior after
changing selection from collapsed to non-collapsed using
`userUpdateTextEditingValue()`.
```

Author: The underlying issue had to do with using multiple sources of truth where one wasn't necessarily up to date (could be one frame behind).

What is clearly missing is "date" as in "up to date" :). Correcting this.

## Timeline

Landed in version: 3.12.0-4.0.pre<br>

> In stable release: 3.13.0

Reviewer: The 3.13 release? It's probably best to just say "TBD" for now.
